### PR TITLE
conditionals-programs_with_decision_points-hard-q8

### PR DESCRIPTION
**Pull request by ketphan02**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [ ]  Are the assertion fail messages good?
    - [ ]  Logical (does it make sense?)
    - [ ]  Readable (can a student understand what’s wrong?)
    - [ ]  Not too feed-y (does it give away the answer?)
    - [ ]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [ ]  Specific (does the fail message relate to the question?)
- [ ]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [ ]  EOF newlines
- [ ]  Do the comments for Parsons follow the correct convention?
- [ ]  Is the question being tested sufficiently?
    - [ ]  Are there representative cases to test the program?
    - [ ]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [ ]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”
---
## Website
- [ ] Is the question text on the Gamification document up to date? 
    > Make sure to include the sample output if there is one for the question

### CODE REVIEW

File: src/conditionals/programs_with_decision_points/hard/q8/PointsScoring.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package conditionals.programs_with_decision_points.hard.q8;
+
+/*
+ * Write a Java Program that asks a player for their score, and prints out the prize that a player wins in a game
+ * depending on their score. If the player scores 5000 points or above, they win a gold prize. If the player scores
+ * between 2500 - 4999 points, they win a silver prize. If the player scores between 1000 - 2499 points, they win a
+ * bronze prize. If they score below 1000 points, they don't win any prize. Make sure to print out an error message if 
+ * the player enters an invalid score.
```
Comment (Hedgemon4) [inline src/conditionals/programs_with_decision_points/hard/q8/PointsScoring.java:8]: I would change the `Make sure to print an error message ... ` sentence because the question does not specify what is a valid score. If we want to make sure the score is >=  0, just change it to `Make sure to check if the score is greater than or equal to 0, and print out an error message if it is not.`

File: src/conditionals/programs_with_decision_points/hard/q8/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+public class MainTest extends BaseTest {
+  // Parsons with distractors
+  public Clause[] testSentence() {
+    TestOption.isInputTest = true;
+    TestOption.defaultInput = "1 1";
```
Comment (Hedgemon4) [inline src/conditionals/programs_with_decision_points/hard/q8/MainTest.java:20]: Why is this the default input?

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/conditionals/programs_with_decision_points/hard/q8/MainTest.java (hunk 2 above)
Comment (ketphan02) [reply src/conditionals/programs_with_decision_points/hard/q8/MainTest.java:20]: My bad :(

File: src/conditionals/programs_with_decision_points/hard/q8/MainTest.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    TestOption.incorrectStructureErrorMessage = "Your program does not print the correct prize for the given score.";
+    runWithInput(score + " ", new Clause[] {
+        new StringLiteral(prize)
+    });
+  }
+
+  @ParameterizedTest
+  @MethodSource("inputProviderInvalid")
+  public void printsErrorMessageWithInvalidInput(int score) throws InvalidClauseException {
+    TestOption.incorrectStructureErrorMessage = "Your program does not print an error message when given invalid score.";
```
Comment (Hedgemon4) [inline src/conditionals/programs_with_decision_points/hard/q8/MainTest.java:67]: Add an to error message (`... when given an invalid score.`)

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

**Review by Hedgemon4 — APPROVED** [review]
Good work
